### PR TITLE
Implement ipfs shutdown command

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by kevina** [comment]
@whyrusleeping is there a reason this is a top-level command:
```
  ipfs daemon start
  ipfs daemon stop
```
makes a lot more sense to me,  for backwards comparability `ipfs daemon` can alias to `ipfs daemon start`

File: core/commands/shutdown.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		Tagline: "Shut down the ipfs daemon",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		nd, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if !nd.OnlineMode() {
```
Comment (kevina) [inline core/commands/shutdown.go]: I am not 100% sure but I don't think this works when the daemon is started with the `--offline` flag.

Comment (kevina) [inline core/commands/shutdown.go]: I am not 100% sure but I don't think this works when the daemon is started with the --offline flag.

File: test/sharness/t0023-shutdown.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+test_launch_ipfs_daemon
+
+test_expect_success "shutdown succeeds" '
+	ipfs shutdown
+'
+
+test_expect_success "daemon no longer running" '
+	test_expect_code  1 kill -0 $IPFS_PID
+'
+
```
Comment (kevina) [inline test/sharness/t0023-shutdown.sh]: I would also test the daemon when started with the `--offline` flag.

When I added the test it failed.

**Comment by whyrusleeping** [comment]
Its a top level command because its rather difficult to make a command be a subcommand of the daemon command. I tried that at first and it would have required me to start refactoring things.

**Comment by whyrusleeping** [comment]
@kevina made it work with --offline. Needed to use the `LocalMode` check instead of the `OnlineMode` one

**Comment by kevina** [comment]
BTW @whyrusleeping codeclimate is failing:

> exported var DaemonShutdownCmd should have comment or be unexported

**Comment by Kubuxu** [comment]
Is there any reason to export this command definition.,w:x

**Comment by whyrusleeping** [comment]
Yeah, probably no reason to export it

**Pull request merged**